backend/amd64/arch.ml

```diff
@@ -73,6 +73,8 @@ type prefetch_info = {
   addr: addressing_mode;
 }
 
+type rounding_mode = Half_to_even | Down | Up | Towards_zero
```

Comment:
Maybe we could change this to be closer to the IEEE rounding mode names:
```
type rounding_mode =
| Ties_to_even
| Toward_negative
| Toward_positive
| Toward_zero
```

Reply:
Or slightly clearer, adding the kind of rounding (nearest vs directed)
```
type rounding_mode =
| Nearest_ties_to_even
| Directed_toward_negative
| Directed_toward_positive
| Directed_toward_zero
```

Reply:
Also, aren't we missing "Current Direction" (use MXCSR.RC rounding mode)?

Reply:
> Also, aren't we missing "Current Direction" (use MXCSR.RC rounding mode)?

I have just added it, for completeness sake. Initially, I thought that the current rounding mode is "round half to even" by default and that we are not supposed to change it in any program, so it would have been covered by existing options.

Reply:
> Or slightly clearer, adding the kind of rounding (nearest vs directed)
> 
> ```
> type rounding_mode =
> | Nearest_ties_to_even
> | Directed_toward_negative
> | Directed_toward_positive
> | Directed_toward_zero
> ```

The naming I used is from Intel's Manual (and also matches `Base.Float.round`), can we keep it as is?

Reply:
Oh I wasn't aware of that. Which manual are you looking at?

This one has a different name for the first mode, but agrees with your naming for the others (Vol. 2B, page 4-565):
https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-instruction-set-reference-manual-325383.pdf

I wouldn't be surprised if there is no consistently used set of names in these manuals....

Reply:
I was referring to the same one. The name of the first mode there is not great, so I used the one from Base, skipping the "nearest" part which seems obvious.

Reply:
Anyway, matching `Base` seems fine, although maybe I should submit a patch to base to use IEEE names.